crates/pipeline_manager/src/integration_test.rs

```diff
@@ -1612,412 +1610,782 @@ async fn pipeline_start_without_compiling() {
-
-    // Update a service using PUT
-    let put_body = json!({
-      "description": "another_example123",
-      "config": {
-        "kafka": {
-          "bootstrap_servers": [
-            "example.com:12",
-          ],
-          "options": {
-            "example3": "value4",
-          }
-        }
-      }
-    });
+
+    // Create second object of type kafka
     let request = config
-        .put(format!("/v0/services/example_kafka_put"), &put_body)
+        .post(
+            "/v0/services",
+            &json!({
+                "name": "object2",
+                "description": "Description of object2",
+                "config": {
+                    "kafka": {
+                        "bootstrap_servers": [
+                            "example"
+                        ],
+                        "options": {
+                            "example2": "value3"
+                        }
+                    },
+                }
+            }),
+        )
         .await;
-    assert_eq!(request.status(), StatusCode::OK);
+    assert_eq!(request.status(), StatusCode::CREATED);
 
-    // Retrieve updated service of PUT and check content
-    let mut request = config
-        .get_ok(format!("/v0/services/example_kafka_put"))
-        .await;
-    let expected_config = KafkaService {
-        bootstrap_servers: vec!["example.com:12".to_string()],
-        options: BTreeMap::from([("example3".to_string(), "value4".to_string())]),
-    };
-    let response: Value = request.json().await.unwrap();
-    let response = response.as_object().unwrap();
-    assert_eq!(response["name"].as_str().unwrap(), "example_kafka_put");
-    assert_eq!(
-        response["description"].as_str().unwrap(),
-        "another_example123"
-    );
-    let v = response["config"].as_object().unwrap()["kafka"].clone();
+    // List of two
+    let mut request = config.get_ok("/v0/services?service_type=kafka").await;
     assert_eq!(
-        serde_json::from_value::<KafkaService>(v).unwrap(),
-        expected_config
+        request
+            .json::<Value>()
+            .await
+            .unwrap()
+            .as_array()
+            .unwrap()
+            .len(),
+        2
     );
-    assert_eq!(response["config_type"].as_str().unwrap(), "kafka");
-
-    // Delete last service
-    let req = config
-        .delete(format!("/v0/services/example_kafka_put"))
-        .await;
-    assert_eq!(StatusCode::OK, req.status(), "Response {:?}", req);
-
-    // Delete non-existing
-    let req = config
-        .delete(format!("/v0/services/example_kafka_put"))
-        .await;
-    assert_eq!(StatusCode::NOT_FOUND, req.status(), "Response {:?}", req);
 
-    // Check no service is left
-    let mut request = config.get_ok(format!("/v0/services")).await;
-    let response: Value = request.json().await.unwrap();
-    assert_eq!(response.as_array().unwrap().len(), 0);
+    // TODO: when new service types are added, some other typed instances should
```

Comment:
this is a lot of work...

Reply:
Yes, the test framework will need some more work, currently will leave out the testing part of this PR and the updates, documentation and comment changes remain.